internal/addrs/check.go

```diff
@@ -1,251 +1,107 @@
 package addrs
 
-import (
-	"fmt"
-
-	"github.com/hashicorp/hcl/v2"
-	"github.com/hashicorp/hcl/v2/hclsyntax"
-	"github.com/hashicorp/terraform/internal/tfdiags"
-)
-
-// Check is the address of a check rule within a checkable object.
-//
-// This represents the check rule globally within a configuration, and is used
-// during graph evaluation to identify a condition result object to update with
-// the result of check rule evaluation.
-//
-// The check address is not distinct from resource traversals, and check rule
-// values are not intended to be available to the language, so the address is
-// not Referenceable.
-//
-// Note also that the check address is only relevant within the scope of a run,
-// as reordering check blocks between runs will result in their addresses
-// changing. Check is therefore for internal use only and should not be exposed
-// in durable artifacts such as state snapshots.
+import "fmt"
+
```

Comment:
It'd be nice to have a godoc comment about what this type means, similar to the one for the now-renamed `CheckRule` type. (Same thought for the other new types in this file.)